web/packages/teleport/src/Console/DocumentDb/ConnectDialog.tsx

```diff
@@ -0,0 +1,202 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import Dialog, {
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from 'design/Dialog';
+import {
+    Box,
+  ButtonPrimary,
+  ButtonSecondary,
+  Flex,
+  Indicator,
+} from 'design';
+
+import Validation from 'shared/components/Validation';
+import { Option } from 'shared/components/Select';
+import { FieldSelect, FieldSelectCreatable } from 'shared/components/FieldSelect';
+import { Database } from 'teleport/services/databases';
+import { useTeleport } from 'teleport';
+import { useUnifiedResourcesFetch } from 'shared/components/UnifiedResources';
+import { Danger } from 'design/Alert';
+import { requiredField } from 'shared/components/Validation/rules';
+
+type Props = {
+  clusterId: string;
+  serviceName: string;
+  onClose(): void;
+  onConnect: onConnectCallback;
+};
+
+type onConnectCallback = (
+  name: string,
+  protocol: string,
+  dbName: string,
+  dbUser: string,
+  dbRoles: string[],
+) => void;
+
+function DbConnectDialog({ clusterId, serviceName, onClose, onConnect }: Props) {
+  // Fetch database information to pre-fill the connection parameters.
+  const ctx = useTeleport();
+  const {
+    fetch: unifiedFetch,
+    attempt,
+    resources,
+  } = useUnifiedResourcesFetch({
+    fetchFunc: useCallback(
+      async (_, signal) => {
+        const response = await ctx.resourceService.fetchUnifiedResources(
+          clusterId,
+          {
+            query: `name == "${serviceName}"`,
+            sort: { fieldName: 'name', dir: 'ASC'},
+            limit: 1,
+          },
+          signal
+        );
+
+
+        // TODO(gabrielcorado): Handle scenarios where there is conflict on the name.
+        if (response.agents.length !== 1 || response.agents[0].kind !== 'db') {
+          throw new Error('Unable to retrieve database information.');
+        }
+
+        return { agents: response.agents };
+      },
+      [clusterId, serviceName]
+    )
+  })
+  useEffect(() => { unifiedFetch({clear: true})}, [])
+
+
+  return (
+    <Dialog
+      dialogCss={dialogCss}
+      disableEscapeKeyDown={false}
+      onClose={onClose}
+      open={true}
+    >
+      <DialogHeader mb={4}>
+        <DialogTitle>Connect To Database</DialogTitle>
+      </DialogHeader>
+
+      {attempt.status === 'failed' && <Danger children={attempt.statusText} />}
+      {(attempt.status === '' || attempt.status === 'processing') && (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      )}
+      {attempt.status === 'success' && <DbConnectDialogForm db={resources[0]} onConnect={onConnect} onClose={onClose} />}
+    </Dialog>
+  );
+}
+
+type FormProps = {
+  db: Database;
+  onConnect: onConnectCallback;
+  onClose(): void;
+};
+
+function DbConnectDialogForm({ db, onConnect, onClose }: FormProps) {
```

Comment:
It's used only in this file, we can use less specific name. Maybe we could call it `ConnectionForm`.